// @vonage/voice/dist/enums/TTSLanguages.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TTSLanguages = void 0;
var TTSLanguages;
(function (TTSLanguages) {
    TTSLanguages["AR"] = "ar";
    TTSLanguages["CA_ES"] = "ca-ES";
    TTSLanguages["CMN_CN"] = "cmn-CN";
    TTSLanguages["CMN_TW"] = "cmn-TW";
    TTSLanguages["CS_CZ"] = "cs-CZ";
    TTSLanguages["CY_GB"] = "cy-GB";
    TTSLanguages["DA_DK"] = "da-DK";
    TTSLanguages["DE_DE"] = "de-DE";
    TTSLanguages["EL_GR"] = "el-GR";
    TTSLanguages["EN_AU"] = "en-AU";
    TTSLanguages["EN_GB"] = "en-GB";
    TTSLanguages["EN_GB_WLS"] = "en-GB-WLS";
    TTSLanguages["EN_IN"] = "en-IN";
    TTSLanguages["EN_US"] = "en-US";
    TTSLanguages["EN_ZA"] = "en-ZA";
    TTSLanguages["ES_ES"] = "es-ES";
    TTSLanguages["ES_MX"] = "es-MX";
    TTSLanguages["ES_US"] = "es-US";
    TTSLanguages["EU_ES"] = "eu-ES";
    TTSLanguages["FI_FI"] = "fi-FI";
    TTSLanguages["FIL_PH"] = "fil-PH";
    TTSLanguages["FR_CA"] = "fr-CA";
    TTSLanguages["FR_FR"] = "fr-FR";
    TTSLanguages["HE_IL"] = "he-IL";
    TTSLanguages["HI_IN"] = "hi-IN";
    TTSLanguages["HU_HU"] = "hu-HU";
    TTSLanguages["ID_ID"] = "id-ID";
    TTSLanguages["IS_IS"] = "is-IS";
    TTSLanguages["IT_IT"] = "it-IT";
    TTSLanguages["JA_JP"] = "ja-JP";
    TTSLanguages["KO_KR"] = "ko-KR";
    TTSLanguages["NB_NO"] = "nb-NO";
    TTSLanguages["NL_NL"] = "nl-NL";
    TTSLanguages["NO_NO"] = "no-NO";
    TTSLanguages["PL_PL"] = "pl-PL";
    TTSLanguages["PT_BR"] = "pt-BR";
    TTSLanguages["PT_PT"] = "pt-PT";
    TTSLanguages["RO_RO"] = "ro-RO";
    TTSLanguages["RU_RU"] = "ru-RU";
    TTSLanguages["SK_SK"] = "sk-SK";
    TTSLanguages["SV_SE"] = "sv-SE";
    TTSLanguages["TH_TH"] = "th-TH";
    TTSLanguages["TR_TR"] = "tr-TR";
    TTSLanguages["UK_UA"] = "uk-UA";
    TTSLanguages["VI_VN"] = "vi-VN";
    TTSLanguages["YUE_CN"] = "yue-CN";
})(TTSLanguages || (exports.TTSLanguages = TTSLanguages = {}));
