// @vonage/voice/dist/enums/TTSVoices.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TTSVoices = void 0;
var TTSVoices;
(function (TTSVoices) {
    TTSVoices["ADITI"] = "Aditi";
    TTSVoices["AGNIESZKA"] = "Agnieszka";
    TTSVoices["ALVA"] = "Alva";
    TTSVoices["AMY"] = "Amy";
    TTSVoices["ASTRID"] = "Astrid";
    TTSVoices["BIANCA"] = "Bianca";
    TTSVoices["BRIAN"] = "Brian";
    TTSVoices["CARLA"] = "Carla";
    TTSVoices["CARMEN"] = "Carmen";
    TTSVoices["CARMIT"] = "Carmit";
    TTSVoices["CATARINA"] = "Catarina";
    TTSVoices["CELINE"] = "Celine";
    TTSVoices["CEM"] = "Cem";
    TTSVoices["CHANTAL"] = "Chantal";
    TTSVoices["CHIPMUNK"] = "Chipmunk";
    TTSVoices["CONCHITA"] = "Conchita";
    TTSVoices["CRISTIANO"] = "Cristiano";
    TTSVoices["DAMAYANTI"] = "Damayanti";
    TTSVoices["DORA"] = "Dora";
    TTSVoices["EMMA"] = "Emma";
    TTSVoices["EMPAR"] = "Empar";
    TTSVoices["ENRIQUE"] = "Enrique";
    TTSVoices["ERIC"] = "Eric";
    TTSVoices["EWA"] = "Ewa";
    TTSVoices["FELIPE"] = "Felipe";
    TTSVoices["FILIZ"] = "Filiz";
    TTSVoices["GERAINT"] = "Geraint";
    TTSVoices["GIORGIO"] = "Giorgio";
    TTSVoices["GWYNETH"] = "Gwyneth";
    TTSVoices["HANS"] = "Hans";
    TTSVoices["HENRIK"] = "Henrik";
    TTSVoices["INES"] = "Ines";
    TTSVoices["IOANA"] = "Ioana";
    TTSVoices["IVETA"] = "Iveta";
    TTSVoices["IVY"] = "Ivy";
    TTSVoices["JACEK"] = "Jacek";
    TTSVoices["JAN"] = "Jan";
    TTSVoices["JENNIFER"] = "Jennifer";
    TTSVoices["JOANA"] = "Joana";
    TTSVoices["JOANNA"] = "Joanna";
    TTSVoices["JOEY"] = "Joey";
    TTSVoices["JORDI"] = "Jordi";
    TTSVoices["JUSTIN"] = "Justin";
    TTSVoices["KANYA"] = "Kanya";
    TTSVoices["KARL"] = "Karl";
    TTSVoices["KENDRA"] = "Kendra";
    TTSVoices["KIMBERLY"] = "Kimberly";
    TTSVoices["LAILA"] = "Laila";
    TTSVoices["LAURA"] = "Laura";
    TTSVoices["LEA"] = "Lea";
    TTSVoices["LEKHA"] = "Lekha";
    TTSVoices["LIV"] = "Liv";
    TTSVoices["LOTTE"] = "Lotte";
    TTSVoices["LUCIA"] = "Lucia";
    TTSVoices["LUCIANA"] = "Luciana";
    TTSVoices["MADS"] = "Mads";
    TTSVoices["MAGED"] = "Maged";
    TTSVoices["MAJA"] = "Maja";
    TTSVoices["MARISKA"] = "Mariska";
    TTSVoices["MARLENE"] = "Marlene";
    TTSVoices["MATHIEU"] = "Mathieu";
    TTSVoices["MATTHEW"] = "Matthew";
    TTSVoices["MAXIM"] = "Maxim";
    TTSVoices["MEI_JIA"] = "Mei-Jia";
    TTSVoices["MELINA"] = "Melina";
    TTSVoices["MIA"] = "Mia";
    TTSVoices["MIGUEL"] = "Miguel";
    TTSVoices["MIREN"] = "Miren";
    TTSVoices["MIZUKI"] = "Mizuki";
    TTSVoices["MONTSERRAT"] = "Montserrat";
    TTSVoices["NAJA"] = "Naja";
    TTSVoices["NICOLE"] = "Nicole";
    TTSVoices["NIKOS"] = "Nikos";
    TTSVoices["NORA"] = "Nora";
    TTSVoices["OSKAR"] = "Oskar";
    TTSVoices["PENELOPE"] = "Penelope";
    TTSVoices["RAVEENA"] = "Raveena";
    TTSVoices["RICARDO"] = "Ricardo";
    TTSVoices["RUBEN"] = "Ruben";
    TTSVoices["RUSSELL"] = "Russell";
    TTSVoices["SALLI"] = "Salli";
    TTSVoices["SATU"] = "Satu";
    TTSVoices["SEOYEON"] = "Seoyeon";
    TTSVoices["SIN_JI"] = "Sin-Ji";
    TTSVoices["SORA"] = "Sora";
    TTSVoices["TAKUMI"] = "Takumi";
    TTSVoices["TARIK"] = "Tarik";
    TTSVoices["TATYANA"] = "Tatyana";
    TTSVoices["TESSA"] = "Tessa";
    TTSVoices["TIAN_TIAN"] = "Tian-Tian";
    TTSVoices["VICKI"] = "Vicki";
    TTSVoices["VITORIA"] = "Vitoria";
    TTSVoices["YELDA"] = "Yelda";
    TTSVoices["ZEINA"] = "Zeina";
    TTSVoices["ZHIYU"] = "Zhiyu";
    TTSVoices["ZUZANA"] = "Zuzana";
})(TTSVoices || (exports.TTSVoices = TTSVoices = {}));
