// @vonage/verify/dist/classes/PSD2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PSD2 = void 0;
class PSD2 {
    number;
    payee;
    amount;
    country;
    codeLength;
    lg;
    /**
       * @deprecated
       */
    language;
    pinExpiry;
    nextEventWait;
    workflowId;
    constructor(phoneNumber, payee, amount, country, codeLength, language, lg, pinExpiry, nextEventWait, workflowId) {
        this.number = phoneNumber;
        if (country) {
            this.country = country;
        }
        if (payee) {
            this.payee = payee;
        }
        if (amount) {
            this.amount = amount;
        }
        if (codeLength) {
            this.codeLength = codeLength;
        }
        if (language) {
            this.lg = language;
            this.language = language;
        }
        if (pinExpiry) {
            this.pinExpiry = pinExpiry;
        }
        if (nextEventWait) {
            this.nextEventWait = nextEventWait;
        }
        if (workflowId) {
            this.workflowId = workflowId;
        }
    }
}
exports.PSD2 = PSD2;
