// @vonage/verify/dist/classes/Verification.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Verification = void 0;
class Verification {
    number;
    brand;
    country;
    senderId;
    codeLength;
    /**
       * @deprecated
       */
    language;
    lg;
    pinExpiry;
    nextEventWait;
    workflowId;
    constructor(phoneNumber, brand, country, senderId, codeLength, language, lg, pinExpiry, nextEventWait, workflowId) {
        this.number = phoneNumber;
        this.brand = brand;
        if (country) {
            this.country = country;
        }
        if (senderId) {
            this.senderId = senderId;
        }
        if (codeLength) {
            this.codeLength = codeLength;
        }
        if (language) {
            this.lg = language;
            this.language = language;
        }
        if (pinExpiry) {
            this.pinExpiry = pinExpiry;
        }
        if (nextEventWait) {
            this.nextEventWait = nextEventWait;
        }
        if (workflowId) {
            this.workflowId = workflowId;
        }
    }
}
exports.Verification = Verification;
