// @vonage/messages/dist/enums/WhatsApp/WhatsAppLanguageCodes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WhatsAppLanguageCode = void 0;
var WhatsAppLanguageCode;
(function (WhatsAppLanguageCode) {
    WhatsAppLanguageCode["AFRIKAANS"] = "af";
    WhatsAppLanguageCode["ALBANIAN"] = "sq";
    WhatsAppLanguageCode["ARABIC"] = "ar";
    WhatsAppLanguageCode["AZERBAIJANI"] = "az";
    WhatsAppLanguageCode["BENGALI"] = "bn";
    WhatsAppLanguageCode["BULGARIAN"] = "bg";
    WhatsAppLanguageCode["CATALAN"] = "ca";
    WhatsAppLanguageCode["CHINESE_CHN"] = "zh_CN";
    WhatsAppLanguageCode["CHINESE_HKG"] = "zh_HK";
    WhatsAppLanguageCode["CHINESE_TAI"] = "zh_TW";
    WhatsAppLanguageCode["CROATIAN"] = "hr";
    WhatsAppLanguageCode["CZECH"] = "cs";
    WhatsAppLanguageCode["DANISH"] = "da";
    WhatsAppLanguageCode["DUTCH"] = "nl";
    WhatsAppLanguageCode["ENGLISH"] = "en";
    WhatsAppLanguageCode["ENGLISH_UK"] = "en_GB";
    WhatsAppLanguageCode["ENGLISH_US"] = "en_US";
    WhatsAppLanguageCode["ESTONIAN"] = "et";
    WhatsAppLanguageCode["FILIPINO"] = "fil";
    WhatsAppLanguageCode["FINNISH"] = "fi";
    WhatsAppLanguageCode["FRENCH"] = "fr";
    WhatsAppLanguageCode["GEORGIAN"] = "ka";
    WhatsAppLanguageCode["GERMAN"] = "de";
    WhatsAppLanguageCode["GREEK"] = "el";
    WhatsAppLanguageCode["GUJARATI"] = "gu";
    WhatsAppLanguageCode["HAUSA"] = "ha";
    WhatsAppLanguageCode["HEBREW"] = "he";
    WhatsAppLanguageCode["HINDI"] = "hi";
    WhatsAppLanguageCode["HUNGARIAN"] = "hu";
    WhatsAppLanguageCode["INDONESIAN"] = "id";
    WhatsAppLanguageCode["IRISH"] = "ga";
    WhatsAppLanguageCode["ITALIAN"] = "it";
    WhatsAppLanguageCode["JAPANESE"] = "ja";
    WhatsAppLanguageCode["KANNADA"] = "kn";
    WhatsAppLanguageCode["KAZAKH"] = "kk";
    WhatsAppLanguageCode["KINYARWANDA"] = "rw_RW";
    WhatsAppLanguageCode["KOREAN"] = "ko";
    WhatsAppLanguageCode["KYRGYZ_KYRGYZSTAN"] = "ky_KG";
    WhatsAppLanguageCode["LAO"] = "lo";
    WhatsAppLanguageCode["LATVIAN"] = "lv";
    WhatsAppLanguageCode["LITHUANIAN"] = "lt";
    WhatsAppLanguageCode["MACEDONIAN"] = "mk";
    WhatsAppLanguageCode["MALAY"] = "ms";
    WhatsAppLanguageCode["MALAYALAM"] = "ml";
    WhatsAppLanguageCode["MARATHI"] = "mr";
    WhatsAppLanguageCode["NORWEGIAN"] = "nb";
    WhatsAppLanguageCode["PERSIAN"] = "fa";
    WhatsAppLanguageCode["POLISH"] = "pl";
    WhatsAppLanguageCode["PORTUGUESE_BR"] = "pt_BR";
    WhatsAppLanguageCode["PORTUGUESE_POR"] = "pt_PT";
    WhatsAppLanguageCode["PUNJABI"] = "pa";
    WhatsAppLanguageCode["ROMANIAN"] = "ro";
    WhatsAppLanguageCode["RUSSIAN"] = "ru";
    WhatsAppLanguageCode["SERBIAN"] = "sr";
    WhatsAppLanguageCode["SLOVAK"] = "sk";
    WhatsAppLanguageCode["SLOVENIAN"] = "sl";
    WhatsAppLanguageCode["SPANISH"] = "es";
    WhatsAppLanguageCode["SPANISH_ARG"] = "es_AR";
    WhatsAppLanguageCode["SPANISH_SPA"] = "es_ES";
    WhatsAppLanguageCode["SPANISH_MEX"] = "es_MX";
    WhatsAppLanguageCode["SWAHILI"] = "sw";
    WhatsAppLanguageCode["SWEDISH"] = "sv";
    WhatsAppLanguageCode["TAMIL"] = "ta";
    WhatsAppLanguageCode["TELUGU"] = "te";
    WhatsAppLanguageCode["THAI"] = "th";
    WhatsAppLanguageCode["TURKISH"] = "tr";
    WhatsAppLanguageCode["UKRAINIAN"] = "uk";
    WhatsAppLanguageCode["URDU"] = "ur";
    WhatsAppLanguageCode["UZBEK"] = "uz";
    WhatsAppLanguageCode["VIETNAMESE"] = "vi";
    WhatsAppLanguageCode["ZULU"] = "zu";
})(WhatsAppLanguageCode || (exports.WhatsAppLanguageCode = WhatsAppLanguageCode = {}));
