// @vonage/verify/dist/enums/VerifyLanguages.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerifyLanguages = void 0;
var VerifyLanguages;
(function (VerifyLanguages) {
    VerifyLanguages["ARABIC"] = "ar-xa";
    VerifyLanguages["CZECH"] = "cs-cz";
    VerifyLanguages["WELSH"] = "cy-cy";
    VerifyLanguages["WELSH_UK"] = "cy-gb";
    VerifyLanguages["DANISH"] = "da-dk";
    VerifyLanguages["GERMAN"] = "de-de";
    VerifyLanguages["GREEK"] = "el-gr";
    VerifyLanguages["ENGLISH_AUSTRALIA"] = "en-au";
    VerifyLanguages["ENGLISH_UK"] = "en-gb";
    VerifyLanguages["ENGLISH_INDIA"] = "en-in";
    VerifyLanguages["ENGLISH_US"] = "en-us";
    VerifyLanguages["SPANISH_SPAIN"] = "es-es";
    VerifyLanguages["SPANISH_MEXICO"] = "es-mx";
    VerifyLanguages["SPANISH_US"] = "es-us";
    VerifyLanguages["FINNISH"] = "fi-fi";
    VerifyLanguages["FILIPINO"] = "fil-ph";
    VerifyLanguages["FRENCH_CANADA"] = "fr-ca";
    VerifyLanguages["FRENCH_FRANCE"] = "fr-fr";
    VerifyLanguages["HINDI"] = "hi-in";
    VerifyLanguages["HUNGARIAN"] = "hu-hu";
    VerifyLanguages["INDONESIAN"] = "id-id";
    VerifyLanguages["ICELANDIC"] = "is-is";
    VerifyLanguages["ITALIAN"] = "it-it";
    VerifyLanguages["JAPANESE"] = "ja-jp";
    VerifyLanguages["KOREAN"] = "ko-kr";
    VerifyLanguages["NORWEGIAN"] = "nb-no";
    VerifyLanguages["DUTCH"] = "nl-nl";
    VerifyLanguages["POLISH"] = "pl-pl";
    VerifyLanguages["PORTUGUESE_BRAZIL"] = "pt-br";
    VerifyLanguages["PORTUGUESE_PORTUGAL"] = "pt-pt";
    VerifyLanguages["ROMANIAN"] = "ro-ro";
    VerifyLanguages["RUSSIAN"] = "ru-ru";
    VerifyLanguages["SWEDISH"] = "sv-se";
    VerifyLanguages["THAI"] = "th-th";
    VerifyLanguages["TURKISH"] = "tr-tr";
    VerifyLanguages["VIETNAMESE"] = "vi-vn";
    VerifyLanguages["YUE_CHINESE_CANTONESE"] = "yue-cn";
    VerifyLanguages["CHINESE_MAINLAND"] = "zh-cn";
    VerifyLanguages["CHINESE_TAIWAN"] = "zh-tw";
})(VerifyLanguages || (exports.VerifyLanguages = VerifyLanguages = {}));
