// @vonage/verify2/dist/verify2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Verify2 = void 0;
const server_client_1 = require("@vonage/server-client");
class Verify2 extends server_client_1.Client {
    authType = server_client_1.AuthenticationType.JWT;
    async newRequest(params) {
        const resp = await this.sendPostRequest(`${this.config.apiHost}/v2/verify`, server_client_1.Client.transformers.snakeCaseObjectKeys(params, true));
        return {
            requestId: resp.data.request_id,
        };
    }
    async checkCode(requestId, code) {
        const resp = await this.sendPostRequest(`${this.config.apiHost}/v2/verify/${requestId}`, {
            code: code,
        });
        return resp.data.status;
    }
    async cancel(requestId) {
        try {
            await this.sendDeleteRequest(`${this.config.apiHost}/v2/verify/${requestId}`);
            return true;
        }
        catch (error) {
            return false;
        }
    }
}
exports.Verify2 = Verify2;
