// @vonage/server-client/dist/enums/GenericErrorCodes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenericErrors = void 0;
var GenericErrors;
(function (GenericErrors) {
    GenericErrors["LOW_BALANCE"] = "low-balance";
    GenericErrors["UNAUTHORIZED"] = "unauthorized";
    GenericErrors["FORBIDDEN"] = "forbidden";
    GenericErrors["NOT_FOUND"] = "not-found";
    GenericErrors["UNPROVISIONED"] = "unprovisioned";
    GenericErrors["ACCOUNT_SUSPENDED"] = "account-suspended";
    GenericErrors["JWT_EXPIRED"] = "jwt-expired";
    GenericErrors["JWT_REVOKED"] = "jwt-revoked";
    GenericErrors["INVALID_API_KEY"] = "invalid-api-key";
    GenericErrors["INVALID_SIGNATURE"] = "invalid-signature";
    GenericErrors["INVALID_IP"] = "invalid-ip";
    GenericErrors["MULTIPLE_AUTH_METHODS"] = "multiple-auth-methods";
    GenericErrors["INVALID_ID"] = "invalid-id";
    GenericErrors["INVALID_JSON"] = "invalid-json";
    GenericErrors["WRONG_VERB"] = "wrong-verb";
    GenericErrors["ACCEPT_HEADER"] = "accept-header";
    GenericErrors["CONTENT_TYPE_HEADER"] = "content-type-header";
    GenericErrors["UNAVAILABLE_LEGAL"] = "unavailable-legal";
    GenericErrors["APPLICATION_SUSPENDED"] = "application-suspended";
})(GenericErrors || (exports.GenericErrors = GenericErrors = {}));
