// @vonage/messages/dist/enums/Messenger/MessengerTags.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessengerTags = void 0;
var MessengerTags;
(function (MessengerTags) {
    MessengerTags["CONFIRMED_EVENT_UPDATE"] = "CONFIRMED_EVENT_UPDATE";
    MessengerTags["POST_PURCHASE_UPDATE"] = "POST_PURCHASE_UPDATE";
    MessengerTags["ACCOUNT_UPDATE"] = "ACCOUNT_UPDATE";
    MessengerTags["HUMAN_AGENT"] = "HUMAN_AGENT";
    MessengerTags["SHIPPING_UPDATE"] = "SHIPPING_UPDATE";
    MessengerTags["RESERVATION_UPDATE"] = "RESERVATION_UPDATE";
    MessengerTags["ISSUE_RESOLUTION"] = "ISSUE_RESOLUTION";
    MessengerTags["APPOINTMENT_UPDATE"] = "APPOINTMENT_UPDATE";
    MessengerTags["GAME_EVENT"] = "GAME_EVENT";
    MessengerTags["TRANSPORTATION_UPDATE"] = "TRANSPORTATION_UPDATE";
    MessengerTags["FEATURE_FUNCTIONALITY_UPDATE"] = "FEATURE_FUNCTIONALITY_UPDATE";
    MessengerTags["TICKET_UPDATE"] = "TICKET_UPDATE";
    MessengerTags["PAYMENT_UPDATE"] = "PAYMENT_UPDATE";
    MessengerTags["PERSONAL_FINANCE_UPDATE"] = "PERSONAL_FINANCE_UPDATE";
})(MessengerTags || (exports.MessengerTags = MessengerTags = {}));
