// @vonage/voice/dist/classes/NCCO/Conversation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Conversation = void 0;
const enums_1 = require("../../enums");
class Conversation {
    action = enums_1.NCCOActions.CONVERSATION;
    name;
    musicOnHoldUrl;
    startOnEnter;
    endOnExit;
    record;
    canSpeak;
    canHear;
    mute;
    constructor(name, musicOnHoldUrl, startOnEnter, endOnExit, record, canSpeak, canHear, mute) {
        this.name = name;
        if (musicOnHoldUrl) {
            this.musicOnHoldUrl = [musicOnHoldUrl];
        }
        if (startOnEnter) {
            this.startOnEnter = startOnEnter;
        }
        if (endOnExit) {
            this.endOnExit = endOnExit;
        }
        if (record) {
            this.record = record;
        }
        if (canSpeak) {
            this.canSpeak = canSpeak;
        }
        if (canHear) {
            this.canHear = canHear;
        }
        if (mute) {
            this.mute = mute;
        }
    }
    serializeToNCCO() {
        const data = {
            action: enums_1.NCCOActions.CONVERSATION,
            name: this.name,
        };
        if (this.musicOnHoldUrl) {
            data.musicOnHoldUrl = this.musicOnHoldUrl;
        }
        if (this.startOnEnter) {
            data.startOnEnter = this.startOnEnter;
        }
        if (this.endOnExit) {
            data.endOnExit = this.endOnExit;
        }
        if (this.record) {
            data.record = this.record;
        }
        if (this.canSpeak) {
            data.canSpeak = this.canSpeak;
        }
        if (this.canHear) {
            data.canHear = this.canHear;
        }
        if (this.mute) {
            data.mute = this.mute;
        }
        return data;
    }
}
exports.Conversation = Conversation;
