// @vonage/server-sdk/dist/vonage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Vonage = void 0;
const accounts_1 = require("@vonage/accounts");
const applications_1 = require("@vonage/applications");
const auth_1 = require("@vonage/auth");
const messages_1 = require("@vonage/messages");
const number_insights_1 = require("@vonage/number-insights");
const numbers_1 = require("@vonage/numbers");
const pricing_1 = require("@vonage/pricing");
const redact_1 = require("@vonage/redact");
const sms_1 = require("@vonage/sms");
const users_1 = require("@vonage/users");
const verify_1 = require("@vonage/verify");
const verify2_1 = require("@vonage/verify2");
const voice_1 = require("@vonage/voice");
const video_1 = require("@vonage/video");
class Vonage {
    credentials;
    options;
    accounts;
    applications;
    messages;
    numberInsights;
    numbers;
    pricing;
    redact;
    secrets;
    sms;
    users;
    verify2;
    verify;
    voice;
    video;
    constructor(credentials, options) {
        if (typeof credentials.getQueryParams === 'undefined') {
            credentials = new auth_1.Auth(credentials);
        }
        this.credentials = credentials;
        this.options = options;
        this.accounts = new accounts_1.Accounts(this.credentials, this.options);
        this.applications = new applications_1.Applications(this.credentials, this.options);
        this.messages = new messages_1.Messages(this.credentials, this.options);
        this.numberInsights = new number_insights_1.NumberInsights(this.credentials, this.options);
        this.numbers = new numbers_1.Numbers(this.credentials, this.options);
        this.pricing = new pricing_1.Pricing(this.credentials, this.options);
        this.redact = new redact_1.Redact(this.credentials, this.options);
        this.secrets = new accounts_1.Secrets(this.credentials, this.options);
        this.sms = new sms_1.SMS(this.credentials, this.options);
        this.users = new users_1.Users(this.credentials, this.options);
        this.verify = new verify_1.Verify(this.credentials, this.options);
        this.verify2 = new verify2_1.Verify2(this.credentials, this.options);
        this.voice = new voice_1.Voice(this.credentials, this.options);
        this.video = new video_1.Video(this.credentials, this.options);
    }
}
exports.Vonage = Vonage;
